pydantic/_internal/_generate_schema.py

```diff
@@ -72,11 +72,13 @@ def generate_config(cls: type[BaseModel]) -> core_schema.CoreConfig:
 
 
 class GenerateSchema:
-    __slots__ = 'arbitrary_types', 'types_namespace'
+    __slots__ = 'arbitrary_types', 'types_namespace', '_seen'
+    _seen: Dict[str, core_schema.RecursiveReferenceSchema]
```

Comment:
I think it would be worth adding a comment here or on the `__init__` describing what this attribute is used for; as of now, you have to read through the typed dict schema logic to understand.

This could also be a good candidate for replacing `str` with a `CoreRef` `NewType` as is done in the `GenerateJsonSchema`, not sure it's worth the effort at this stage though.